webpack/assets/javascripts/react_app/components/bookmarks/index.js

```diff
@@ -12,13 +12,15 @@ import helpers from '../../common/helpers';
 class BookmarkContainer extends React.Component {
   constructor(props) {
     super(props);
-    helpers.bindMethods(this, ['handleNewBookmarkClick']);
+    helpers.bindMethods(this, ['handleNewBookmarkClick', 'loadBookmarks']);
   }
 
-  componentDidMount() {
-    const { url, controller, getBookmarks } = this.props;
+  loadBookmarks() {
+    if (this.props.bookmarks.length === 0 && this.props.status !== STATUS.PENDING) {
```

Comment:
I am afraid the logic under `if` will grow. 
what about:

```js
const {bookmarks, status, url, controller, getBookmarks} = this.props;
if (bookmarks.length > 0 || status === STATUS.PENDING) {
  // do nothing: bookmarks are being loaded or already loaded earlier
  return ;
}
getBookmarks(url, controller)
```

In addition, should we add tests?

Other than that LGTM :+1:

Reply:
a. I agree the statement might grow, but if it does, we will change as needed?
b. I didnt extract the values just to reduce the amount of const extractions in case its a noop - don't mind changing but ymmv
c. regarding tests, whats the best way to test (now that I'm a bit pushed away from the code)? I can see test that onclick calls the function, and that the condition are tested (e.g. if the action is fired or not)?

Reply:
I tried to write a test to `BookmarksContainer` but I realized that the class itself isn't exported and thus getting access to `getBookmarks`/`loadBookmarks` is not trivial as I though. :open_mouth: 

Regarding a and b - you're probably right. :smiley:

Reply:
@ripcurld any suggestions ?

Reply:
nope but don't want to block this either. I tested it manually and it works :+1: .

If I figure a way to test it, I will send a pull request.

Reply:
@ohadlevy I think I worked it out. See [here](https://github.com/ripcurld/foreman/commit/c713d966b245408729147c7a0b83e67b80c6e82e).

Let me know what you think.